### PR TITLE
Download repo contents upon receiving push event from github

### PR DESCRIPTION
**Pull request by hjaensch7**

### CODE REVIEW

**Comment by Others** [comment]
You pushed our secret again... move it to a git-ignored file and load it on startup

File: main.go [hunk 1]
```diff
@@ -34,3 +36,16 @@ func main() {
 		fmt.Println("http.http.ListenAndServe Error", err)
 	}
 }
+
+func getHTTPDownloadURL(p github.PushPayload) string {
+	return "git::" + p.Repository.URL
+}
+
+func downloadRepo(downloadURL string) {
+	err := getter.Get("./repo", downloadURL)
```
Comment (Others) [inline main.go]: Maybe give this function a parameter indicating where to put the download

File: main.go [hunk 2]
```diff
@@ -11,21 +12,22 @@ const (
...
-			fmt.Println("github payload parse failed", err)
+			fmt.Println("github payload parse error:", err)
 			return
 		}
 		fmt.Println("Received Payload:")
 		push := payload.(github.PushPayload)
-		fmt.Println("PUSH PAYLOAD:")
-		fmt.Println(push)
+		downloadURL := getHTTPDownloadURL(push)
+		fmt.Println("Download URL: ", downloadURL)
```
Comment (Others) [inline main.go]: Does ‘Println’ already insert a space between parameters?

File: main.go [hunk 3]
```diff
@@ -34,3 +36,16 @@ func main() {
...
+
+func getHTTPDownloadURL(p github.PushPayload) string {
+	return "git::" + p.Repository.URL
+}
+
+func downloadRepo(downloadURL string) {
+	err := getter.Get("./repo", downloadURL)
+	if err != nil {
+		fmt.Println("Error downloading repo:", err)
+		return
```
Comment (Others) [inline main.go]: Propagate the error

File: main.go [hunk 4]
```diff
@@ -34,3 +36,16 @@ func main() {
...
+	}
+	fmt.Println("Repo downloaded from:", downloadURL)
```
Comment (Others) [inline main.go]: Should we use a better logging library?

**Review by Others — CHANGES_REQUESTED** [review]
Looks basically good, just a few small things

**Comment by Others** [comment]
@hjaensch7 any progress on this?

File: main.go (hunk 1 above)
Comment (hjaensch7) [reply main.go]: this makes sense, so I added it

File: main.go (hunk 2 above)
Comment (hjaensch7) [reply main.go]: yes, yes it does

File: main.go (hunk 4 above)
Comment (hjaensch7) [reply main.go]: switching to the standard go log package

File: main.go [hunk 5]
```diff
@@ -1,36 +1,82 @@
...
 
 const (
 	path = "/payload"
+	port = ":3069"
 )
 
+var (
+	Info    *log.Logger
+	Warning *log.Logger
+	Error   *log.Logger
```
Comment (Others) [inline main.go:20]: Is upper case conventional?

**Review by Others — APPROVED** [review]

**Pull request merged**